smoke-tests/framework/src/main/java/com/microsoft/applicationinsights/smoketest/fakeingestion/MockedAppInsightsIngestionServer.java

```diff
@@ -201,13 +201,35 @@ public List<Envelope> waitForMetricItems(String metricName, int numItems)
     return waitForMetricItems(metricName, numItems, 10, TimeUnit.SECONDS);
   }
 
+  public List<Envelope> waitForMetricItems(
+      String metricName, String secondPredicate, int numItems, boolean isRolename)
+      throws InterruptedException, TimeoutException {
+    return waitForMetricItems(
+        metricName, secondPredicate, numItems, 10, TimeUnit.SECONDS, isRolename);
+  }
+
   public List<Envelope> waitForMetricItems(
       String metricName, int numItems, int timeout, TimeUnit timeUnit)
       throws InterruptedException, TimeoutException {
     return servlet.waitForItems(
         SmokeTestExtension.getMetricPredicate(metricName), numItems, timeout, timeUnit);
   }
 
+  public List<Envelope> waitForMetricItems(
+      String metricName,
+      String secondPredicate,
+      int numItems,
+      int timeout,
+      TimeUnit timeUnit,
+      boolean isRolename)
+      throws InterruptedException, TimeoutException {
+    return servlet.waitForItems(
+        SmokeTestExtension.getMetricPredicate(metricName, secondPredicate, isRolename),
+        numItems,
+        timeout,
+        timeUnit);
+  }
```

Comment:
same with this, I'm wondering if it can just be part of the one smoke test, instead of in the reusable component

Reply:
same here